src/argilla/server/schemas/v1/datasets.py

```diff
@@ -621,12 +621,44 @@ def metadata_parsed(self) -> List[MetadataParsedQueryParam]:
         return [MetadataParsedQueryParam(q) for q in self.metadata]
 
 
-class TextQuery(BaseModel):
-    text: StringQuery
+class VectorQuery(BaseModel):
+    name: str
+
+    record_id: Optional[UUID] = None
+    value: Optional[List[float]] = None
+
+    order: Union[Literal["most_similar"], Literal["least_similar"]] = "most_similar"
+
+    @root_validator
+    def check_required(cls, values: dict) -> dict:
+        """Check that either 'record_id' or 'value' is provided"""
+        record_id = values.get("record_id")
+        value = values.get("value")
+
+        if record_id is None and value is None:
```

Comment:
I was thinking that we must check that one of the two attributes are present but also raise an error when the two of them are present (equivalent to a logical xor condition).